miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/page/BulkSamplePage.java

```diff
@@ -74,4 +106,36 @@ public HandsOnTable getTable() {
     return table;
   }
 
+  public void clickSaveButton() {
+    saveButton.click();
+    // give time for spinner to appear, then wait for it to be hidden again
+    waitExplicitly(1000);
+    waitWithTimeout().until(invisibilityOf(ajaxSpinner));
+  }
+
+  public String getSuccessMessages() {
+    return successMessages.getText();
+  }
+
+  public String getErrorMessages() {
+    return errors.getText();
+  }
+
+  public boolean areErrorsHidden() {
+    return errors.getAttribute("class").contains("hidden");
+  }
+
+  public void waitForIdentityLookup() {
+    waitExplicitly(3000);
```

Comment:
No luck with a conditional wait, as all the text matcher conditionals take in a `By` element or a `WebElement` to locate the thing to match. And we don't want to match the entire table...